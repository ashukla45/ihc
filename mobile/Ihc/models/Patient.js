import {stringDate} from '../util/Date';
export default class Patient {
  // Insert any class methods here
  get drugUpdates() {
    const drugToUpdates = {};
    for (var update of this.medications) {
      if (update.name in drugToUpdates) {
        drugToUpdates[update.name].push(update);
      } else{
        drugToUpdates[update.name] = [update];
      }
    }
    return drugToUpdates;
  }

  get age() {
    const strBirthday = this.birthday;
    const today = stringDate(new Date());
    const birthdayYear = parseInt(strBirthday.slice(0,4)); // Get the year
    const birthdayMonthDay= parseInt(strBirthday.slice(4));
    const todayYear = parseInt(today.slice(0,4));
    const todayMonthDay = parseInt(today.slice(4));

    let age = todayYear - birthdayYear;
    // If hasn't passed their birthday this year, then subtract
    if(birthdayMonthDay < todayMonthDay) {
      age--;
    }
    return age;
  }

  /* Returns {
   *   weights: [[ageInMonths, weight (kg)],...]},
   *   heights: [[ageInMonths, height (cm)],...]}
   * }
   */
  get growthChartData() {
    const triages = this.triages;
    const weights = [];
    const heights = [];
    triages.forEach( triage => {
      const ageInMonths = triage.age * 12; // Rounds off to whole year, should be fine
      weights.push([ageInMonths, triage.weight]);
      heights.push([ageInMonths, triage.height]);
    });
    return {weights, heights};
  }

  get isMale() {
    return this.gender === 1;
  }

  static fullName(patient) {
    return `${patient.firstName} ${patient.fatherName} ${patient.motherName}`;
  }

  // To be used as primary key
  static makeKey(patient) {
    const str = `${patient.firstName}&${patient.fatherName}&${patient.motherName}` +
      `&${patient.birthday}`;
    return str;
  }

  static extractFromForm(form) {
    const patient = Object.assign({}, form);
    patient.birthday = stringDate(form.birthday);
    patient.key = Patient.makeKey(patient);
    patient.needToUpload = false;
    if(form.newPatient) {
      // 1 is male, 2 is female
      patient.gender = form.gender === 'Male' ? 1 : 2;
    }
    return patient;
  }

  // Can pass in parameters to override defaults, mostly useful for tests
  static getInstance(lastUpdated = new Date().getTime(),
    key = 'firstname&father&mother&20000101', firstName = 'firstname',
    fatherName = 'father', motherName = 'mother', birthday = '20000101',
    gender = 1, phone = null, motherHeight = 100, fatherHeight = 100,
    medications = [], soaps = [], triages = [], statuses = []) {
    return {
      key, firstName, fatherName, motherName, birthday, gender, phone, motherHeight,
      fatherHeight, medications, soaps, triages, statuses, lastUpdated
    };
  }
}

Patient.schema = {
  name: 'Patient',
  primaryKey: 'key',
  properties: {
    key: 'string',
    firstName: 'string',
    fatherName: 'string', // last name
    motherName: 'string', // last name
    birthday: 'string',
    gender: 'int', // 1 = boy, 2 = girl, 0 = undefined
    phone: 'string?',
    motherHeight: 'double?',
    fatherHeight: 'double?',
    medications: 'DrugUpdate[]',
    soaps: 'Soap[]',
    triages: 'Triage[]',
    statuses: 'Status[]',
    lastUpdated: 'int', // timestamp
    needToUpload: 'bool'
  }
};
