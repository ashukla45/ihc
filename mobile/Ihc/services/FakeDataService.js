/* eslint-disable */
// This file should hold skeletons for the function calls from RealDataService.
// Implementation is not required because they should be stubbed out during
// tests
export function getPatient(patientKey) {
  return {};
}

export function getTriage(patientKey, date) {
  return {};
}

export function getSoap(patientKey, date) {
  return {};
}

export function getPatients(param) {
  return []; // Return whatever sample data you want
}

export function getUpdates(param) {
  return []; // Return whatever sample data you want
}

/*
 * active_only is true if we only want the statuses of patients that are
 * currently checked in
 */
export function getPatientSelectRows() {
  const arr =
    [
      newFakePatientSelectRow('Jose'),
      newFakePatientSelectRow('Bob'),
      newFakePatientSelectRow('Matt'),
      newFakePatientSelectRow('Coco'),
      newFakePatientSelectRow('Brandon'),
      newFakePatientSelectRow('Evan'),
      newFakePatientSelectRow('Sarah'),
      newFakePatientSelectRow('Daniel'),
      newFakePatientSelectRow('Neil'),
      newFakePatientSelectRow('William')
    ];

  const columnOrder = ['name', 'birthday', 'checkin_time', 'triage_completed',
    'doctor_completed', 'pharmacy_completed', 'notes'];
  const toReturn = createMatrix(arr, columnOrder);

  return new Promise((resolve, reject) => {
    setTimeout(resolve, 100, toReturn);
  });
}

export function getMedicationUpdates() {
  const arr = [
    newFakeMedicationUpdate('Tylenol'),
    newFakeMedicationUpdate('Tylenol'),
    newFakeMedicationUpdate('Advil'),
    newFakeMedicationUpdate('Aspirin'),
    newFakeMedicationUpdate('Aspirin'),
    newFakeMedicationUpdate('Aspirin'),
    newFakeMedicationUpdate('Cold meds'),
  ];

  return new Promise((resolve, reject) => {
    setTimeout(resolve, 100, arr);
  });
}

const medsCount = 0;
function newFakeMedicationUpdate(name) {
  medsCount = ++medsCount % 3;

  return {
    name: name,
    date: `2018020${medsCount}`,
    dose: '10mg',
    frequency: 'bid',
    duration: '1 month',
    notes: 'yes'
  };
}

const numFakePatients = 0;
// Currently ignore triage/doctor/pharmacy params
function newFakePatientSelectRow(name, triage = false, doctor = false,
  pharm = false, notes = '') {
  numFakePatients++;
  return {
    name: name,
    birthday: '20180201',
    checkin_time: new Date().getTime(),
    triage_completed: numFakePatients % 2 === 0 ? new Date().getTime() : null,
    doctor_completed: numFakePatients % 4 === 0 ? new Date().getTime() : null,
    pharmacy_completed: numFakePatients % 8 === 0 ? new Date().getTime() : null,
    notes: notes
  };
}

function createMatrix(data, columnOrder) {
  return data.map((obj) => columnOrder.map( (key) => obj[key] ));
}
/* eslint-enable */
