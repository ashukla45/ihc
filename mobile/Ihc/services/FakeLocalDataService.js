/* eslint-disable */
// This file should hold skeletons for the function calls from RealDataService.
// Implementation is not required because they should be stubbed out during
// tests
export function createPatient(patient) {
}

export function signinPatient(patientForm) {
}

export function updateStatus(patientKey, strDate, field, value) {
}

export function createDrugUpdate(update) {
}

export function updateSoap(update) {
}

export function getSoap(patientKey, strDate) {
}

export function updateTriage(update) {
}

export function getTriage(patientKey, strDate) {
}

export function getPatient(patientKey) {
}

export function getPatientsToUpload() {
}

export function lastSynced() {
}

export function handleDownloadedPatients(patients) {
}

/*
 * active_only is true if we only want the statuses of patients that are
 * currently checked in
 */
export function getPatientSelectRows() {
  const arr =
    [
      newFakePatientSelectRow('Jose'),
      newFakePatientSelectRow('Bob'),
      newFakePatientSelectRow('Matt'),
      newFakePatientSelectRow('Coco'),
      newFakePatientSelectRow('Brandon'),
      newFakePatientSelectRow('Evan'),
      newFakePatientSelectRow('Sarah'),
      newFakePatientSelectRow('Daniel'),
      newFakePatientSelectRow('Neil'),
      newFakePatientSelectRow('William')
    ];

  const columnOrder = ['name', 'birthday', 'checkin_time', 'triage_completed',
    'doctor_completed', 'pharmacy_completed', 'notes'];
  const toReturn = createMatrix(arr, columnOrder);

  return new Promise((resolve, reject) => {
    setTimeout(resolve, 100, toReturn);
  });
}

export function getMedicationUpdates() {
  const arr = [
    newFakeMedicationUpdate('Tylenol'),
    newFakeMedicationUpdate('Tylenol'),
    newFakeMedicationUpdate('Advil'),
    newFakeMedicationUpdate('Aspirin'),
    newFakeMedicationUpdate('Aspirin'),
    newFakeMedicationUpdate('Aspirin'),
    newFakeMedicationUpdate('Cold meds'),
  ];

  return new Promise((resolve, reject) => {
    setTimeout(resolve, 100, arr);
  });
}

const medsCount = 0;
function newFakeMedicationUpdate(name) {
  medsCount = ++medsCount % 3;

  return {
    name: name,
    date: `2018020${medsCount}`,
    dose: '10mg',
    frequency: 'bid',
    duration: '1 month',
    notes: 'yes'
  };
}

const numFakePatients = 0;
// Currently ignore triage/doctor/pharmacy params
function newFakePatientSelectRow(name, triage = false, doctor = false,
  pharm = false, notes = '') {
  numFakePatients++;
  return {
    name: name,
    birthday: '20180201',
    checkin_time: new Date().getTime(),
    triage_completed: numFakePatients % 2 === 0 ? new Date().getTime() : null,
    doctor_completed: numFakePatients % 4 === 0 ? new Date().getTime() : null,
    pharmacy_completed: numFakePatients % 8 === 0 ? new Date().getTime() : null,
    notes: notes
  };
}

function createMatrix(data, columnOrder) {
  return data.map((obj) => columnOrder.map( (key) => obj[key] ));
}
/* eslint-enable */
