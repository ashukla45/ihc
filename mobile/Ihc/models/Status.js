import {stringDate} from '../util/Date';
import Patient from './Patient';
/*
 * A patient's status as they go through their appointment
 */
export default class Status {
  // Insert any class methods here

  static newStatus(patient) {
    const obj = {
      patientKey: patient.key,
      name: Patient.fullName(patient),
      birthday: patient.birthday,
      date: stringDate(new Date()),
      active: true,
      checkinTime: new Date().getTime(),
      lastUpdated: new Date().getTime(),
    };
    return obj;
  }
}

Status.schema = {
  name: 'Status',
  properties: {
    patientKey: 'string',
    name: 'string',
    birthday: 'string', // For convenience in patient select table
    date: 'string',
    active: 'bool',
    checkinTime: 'int', // Timestamp, milliseconds
    triageCompleted: 'int?', // timestamp for when completed
    doctorCompleted: 'int?',
    pharmacyCompleted: 'int?',
    notes: 'string?',
    lastUpdated: 'int',
  }
};
