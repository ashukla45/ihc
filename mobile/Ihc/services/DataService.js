/* global */
// Returns either fakeLocalData or RealData depending on config file
// Config file:
// "FakeDataServices": "true" or "false"
import * as fakeLocalData from '../services/FakeLocalDataService';
import * as localDataService from '../services/LocalDataService';

import * as fakeServerData from '../services/FakeServerDataService';
import * as serverDataService from '../services/ServerDataService';

import config from '../config.json';

// Dont use the Fakes in tests, so that we can test the LocalDataService itself,
// but should stub out its methods for the other tests
let localData = (config.fakeLocalDataService === 'true') ? fakeLocalData : localDataService;

// If we are testing the server data service, then use the real one
let serverData = (config.fakeServerDataService === 'true' && config.testingServerDataService == 'false')
  ? fakeServerData : serverDataService;

export {localData, serverData};
