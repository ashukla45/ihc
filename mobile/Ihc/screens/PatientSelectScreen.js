import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  ScrollView,
} from 'react-native';
import {localData, serverData} from '../services/DataService';
import {stringDate} from '../util/Date';
import PatientTable from '../components/PatientTable';
import Container from '../components/Container';

export default class PatientSelectScreen extends Component<{}> {
  constructor(props) {
    super(props);

    this.state = {
      errorMsg: null,
      successMsg: null,
      loading: false,
      rows: [],
      showRetryButton: false,
    };
    this.props.navigator.setOnNavigatorEvent(this.onNavigatorEvent.bind(this));
  }

  loadPatients = () => {
    this.setState({ loading: true });
    const data = localData.getPatientSelectRows();
    this.setState({ rows: data, loading: false });
  }

  // Reload table after moving back to table
  onNavigatorEvent(event) {
    if (event.id === 'willAppear') {
      this.loadPatients();
    }
  }

  componentDidMount() {
    this.loadPatients();
  }

  goToPatient = (patient) => {
    this.props.navigator.push({
      screen: 'Ihc.PatientHomeScreen',
      title: patient.name,
      passProps: { name: patient[0], patientKey: patient[7] }
    });
  }

  // Update the statusObj with notes from the modal
  saveModal = (patientKey, notes) => {
    let statusObj = {};
    try {
      statusObj = localData.updateStatus(patientKey, stringDate(new Date()),
        'notes', notes);
    } catch(e) {
      this.setState({errorMsg: e.message, successMsg: null});
      return;
    }

    this.setState({loading: true});
    serverData.updateStatus(statusObj)
      .then( () => {
        if(this.state.loading) {
          // if successful, then reload data and close modal
          this.loadPatients();
          this.setState({
            loading: false,
            showRetryButton: false,
            successMsg: 'Saved successfully',
            errorMsg: null
          });
          this.closeModal();
        }
      })
      .catch( (e) => {
        if(this.state.loading) {
          localData.markPatientNeedToUpload(patientKey);
          this.setState({
            errorMsg: e.message,
            successMsg: null,
            loading: false,
            showRetryButton: true
          });
        }
      });
  }

  // If Loading was canceled, we want to show a retry button
  setLoading = (val, canceled) => {
    this.setState({loading: val, showRetryButton: canceled});
  }

  setMsg = (type, msg) => {
    const obj = {};
    obj[type] = msg;
    const other = type === 'successMsg' ? 'errorMsg' : 'successMsg';
    obj[other] = null;
    this.setState(obj);
  }

  render() {
    return (
      <Container loading={this.state.loading}
        errorMsg={this.state.errorMsg}
        successMsg={this.state.successMsg}
        setLoading={this.setLoading}
        setMsg={this.setMsg}
        patientKey={this.state.patientKey}
        showRetryButton={this.state.showRetryButton}
      >
        <Text style={styles.title}>
          Select a Patient
        </Text>

        <ScrollView contentContainerStyle={styles.tableContainer} horizontal>
          <PatientTable
            rows={this.state.rows}
            goToPatient={this.goToPatient}
            saveModal={this.saveModal}
          />
        </ScrollView>
      </Container>
    );
  }
}

const styles = StyleSheet.create({
  title: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  tableContainer: {
    width: '100%'
  },
});
