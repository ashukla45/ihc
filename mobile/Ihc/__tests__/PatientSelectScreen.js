import 'react-native';
import React from 'react';
import PatientSelectScreen from '../screens/PatientSelectScreen';

import Patient from '../models/Patient';
import Status from '../models/Status';
import {localData} from '../services/DataService';

import sinon from 'sinon';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  sinon.useFakeTimers(100);
  const patient = Patient.getInstance();
  const patient2 = Patient.getInstance();
  const statuses = [Status.newStatus(patient), Status.newStatus(patient2)];
  const columnOrder = ['name', 'birthday', 'checkinTime', 'triageCompleted',
    'doctorCompleted', 'pharmacyCompleted', 'notes', 'patientKey'];

  const rows = statuses.map((obj) => columnOrder.map( (key) => obj[key] ));

  sinon.stub(localData, 'getPatientSelectRows').returns(rows);

  const fakeNavigator = { setOnNavigatorEvent: (param) => {} };
  const json = renderer.create(
    <PatientSelectScreen navigator={fakeNavigator} />
  ).toJSON();
  expect(json).toMatchSnapshot();
});
