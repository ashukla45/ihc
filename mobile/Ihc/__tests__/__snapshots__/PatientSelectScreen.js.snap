// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F5FCFF",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "flex": 0,
        "justifyContent": "center",
        "padding": 20,
      }
    }
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "margin": 10,
            "textAlign": "center",
          }
        }
      >
        Select a Patient
      </Text>
      <RCTScrollView
        contentContainerStyle={
          Object {
            "width": "100%",
          }
        }
        horizontal={true}
      >
        <View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Modal
              animationType="fade"
              hardwareAccelerated={false}
              onRequestClose={[Function]}
              transparent={true}
              visible={false}
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#f6fdff",
                      "borderRadius": 8,
                      "borderWidth": 2,
                      "height": "60%",
                      "justifyContent": "center",
                      "width": "80%",
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "fontSize": 24,
                        "textAlign": "center",
                      }
                    }
                  />
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                  >
                    Notes:
                  </Text>
                  <TextInput
                    allowFontScaling={true}
                    multiline={true}
                    numberOfLines={4}
                    onChangeText={[Function]}
                    style={
                      Object {
                        "backgroundColor": "white",
                        "borderRadius": 8,
                        "borderWidth": 1,
                        "margin": 8,
                        "width": "90%",
                      }
                    }
                    value={null}
                  />
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "flex": 1,
                        "flexDirection": "row",
                        "height": 60,
                        "justifyContent": "center",
                        "margin": 4,
                      }
                    }
                  >
                    <View
                      accessibilityComponentType={undefined}
                      accessibilityLabel={undefined}
                      accessibilityTraits={undefined}
                      accessible={true}
                      collapsable={undefined}
                      hasTVPreferredFocus={undefined}
                      hitSlop={undefined}
                      isTVSelectable={true}
                      nativeID={undefined}
                      onLayout={undefined}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "backgroundColor": "#2196F3",
                          "borderRadius": 2,
                          "elevation": 4,
                          "height": 40,
                          "margin": 10,
                          "opacity": 1,
                          "padding": 8,
                          "width": 150,
                        }
                      }
                      testID={undefined}
                      tvParallaxProperties={undefined}
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Object {
                            "color": "#fefefe",
                            "fontWeight": "500",
                            "textAlign": "center",
                          }
                        }
                      >
                        Cancel
                      </Text>
                    </View>
                    <View
                      accessibilityComponentType={undefined}
                      accessibilityLabel={undefined}
                      accessibilityTraits={undefined}
                      accessible={true}
                      collapsable={undefined}
                      hasTVPreferredFocus={undefined}
                      hitSlop={undefined}
                      isTVSelectable={true}
                      nativeID={undefined}
                      onLayout={undefined}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "backgroundColor": "#2196F3",
                          "borderRadius": 2,
                          "elevation": 4,
                          "height": 40,
                          "margin": 10,
                          "opacity": 1,
                          "padding": 8,
                          "width": 150,
                        }
                      }
                      testID={undefined}
                      tvParallaxProperties={undefined}
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Object {
                            "color": "#fefefe",
                            "fontWeight": "500",
                            "textAlign": "center",
                          }
                        }
                      >
                        Save
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </Modal>
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignSelf": "stretch",
                    "backgroundColor": "#dbdbdb",
                    "borderWidth": 1,
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  size={2}
                  style={
                    Object {
                      "borderWidth": 1,
                      "flex": 2,
                      "flexDirection": "column",
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "textAlign": "center",
                      }
                    }
                  >
                    Name
                  </Text>
                </View>
                <View
                  size={1.5}
                  style={
                    Object {
                      "borderWidth": 1,
                      "flex": 1.5,
                      "flexDirection": "column",
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "textAlign": "center",
                      }
                    }
                  >
                    Birthday
                  </Text>
                </View>
                <View
                  size={1}
                  style={
                    Object {
                      "borderWidth": 1,
                      "flex": 1,
                      "flexDirection": "column",
                      "maxWidth": 60,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "textAlign": "center",
                      }
                    }
                  >
                    Checkin
                  </Text>
                </View>
                <View
                  size={1}
                  style={
                    Object {
                      "borderWidth": 1,
                      "flex": 1,
                      "flexDirection": "column",
                      "maxWidth": 60,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "textAlign": "center",
                      }
                    }
                  >
                    Triage
                  </Text>
                </View>
                <View
                  size={1}
                  style={
                    Object {
                      "borderWidth": 1,
                      "flex": 1,
                      "flexDirection": "column",
                      "maxWidth": 60,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "textAlign": "center",
                      }
                    }
                  >
                    Doctor
                  </Text>
                </View>
                <View
                  size={1}
                  style={
                    Object {
                      "borderWidth": 1,
                      "flex": 1,
                      "flexDirection": "column",
                      "maxWidth": 60,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "textAlign": "center",
                      }
                    }
                  >
                    Pharmacy
                  </Text>
                </View>
                <View
                  size={3}
                  style={
                    Object {
                      "borderWidth": 1,
                      "flex": 3,
                      "flexDirection": "column",
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "textAlign": "center",
                      }
                    }
                  >
                    Notes
                  </Text>
                </View>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignSelf": "stretch",
                    "flex": 1,
                    "flexDirection": "row",
                    "minHeight": 32,
                    "opacity": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  onPress={[Function]}
                  style={
                    Object {
                      "alignSelf": "stretch",
                      "flex": 1,
                      "flexDirection": "row",
                      "minHeight": 32,
                    }
                  }
                >
                  <View
                    size={2}
                    style={
                      Object {
                        "borderWidth": 1,
                        "flex": 2,
                        "flexDirection": "column",
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                    >
                      firstname father mother
                    </Text>
                  </View>
                  <View
                    size={1.5}
                    style={
                      Object {
                        "borderWidth": 1,
                        "flex": 1.5,
                        "flexDirection": "column",
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                    >
                      01/01/00
                    </Text>
                  </View>
                  <View
                    size={1}
                    style={
                      Object {
                        "borderWidth": 1,
                        "flex": 1,
                        "flexDirection": "column",
                        "maxWidth": 60,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                    >
                      16:0
                    </Text>
                  </View>
                  <View
                    size={1}
                    style={
                      Object {
                        "borderWidth": 1,
                        "flex": 1,
                        "flexDirection": "column",
                        "maxWidth": 60,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                    >
                      
                    </Text>
                  </View>
                  <View
                    size={1}
                    style={
                      Object {
                        "borderWidth": 1,
                        "flex": 1,
                        "flexDirection": "column",
                        "maxWidth": 60,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                    >
                      
                    </Text>
                  </View>
                  <View
                    size={1}
                    style={
                      Object {
                        "borderWidth": 1,
                        "flex": 1,
                        "flexDirection": "column",
                        "maxWidth": 60,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                    >
                      
                    </Text>
                  </View>
                  <View
                    size={3}
                    style={
                      Object {
                        "borderWidth": 1,
                        "flex": 3,
                        "flexDirection": "column",
                      }
                    }
                  >
                    <View
                      accessibilityComponentType={undefined}
                      accessibilityLabel={undefined}
                      accessibilityTraits={undefined}
                      accessible={true}
                      collapsable={undefined}
                      hasTVPreferredFocus={undefined}
                      hitSlop={undefined}
                      isTVSelectable={true}
                      nativeID={undefined}
                      onLayout={undefined}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "backgroundColor": "#bebebe",
                          "height": "100%",
                          "opacity": 1,
                          "width": "100%",
                        }
                      }
                      testID={undefined}
                      tvParallaxProperties={undefined}
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                      />
                    </View>
                  </View>
                </View>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignSelf": "stretch",
                    "flex": 1,
                    "flexDirection": "row",
                    "minHeight": 32,
                    "opacity": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  onPress={[Function]}
                  style={
                    Object {
                      "alignSelf": "stretch",
                      "flex": 1,
                      "flexDirection": "row",
                      "minHeight": 32,
                    }
                  }
                >
                  <View
                    size={2}
                    style={
                      Object {
                        "borderWidth": 1,
                        "flex": 2,
                        "flexDirection": "column",
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                    >
                      firstname father mother
                    </Text>
                  </View>
                  <View
                    size={1.5}
                    style={
                      Object {
                        "borderWidth": 1,
                        "flex": 1.5,
                        "flexDirection": "column",
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                    >
                      01/01/00
                    </Text>
                  </View>
                  <View
                    size={1}
                    style={
                      Object {
                        "borderWidth": 1,
                        "flex": 1,
                        "flexDirection": "column",
                        "maxWidth": 60,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                    >
                      16:0
                    </Text>
                  </View>
                  <View
                    size={1}
                    style={
                      Object {
                        "borderWidth": 1,
                        "flex": 1,
                        "flexDirection": "column",
                        "maxWidth": 60,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                    >
                      
                    </Text>
                  </View>
                  <View
                    size={1}
                    style={
                      Object {
                        "borderWidth": 1,
                        "flex": 1,
                        "flexDirection": "column",
                        "maxWidth": 60,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                    >
                      
                    </Text>
                  </View>
                  <View
                    size={1}
                    style={
                      Object {
                        "borderWidth": 1,
                        "flex": 1,
                        "flexDirection": "column",
                        "maxWidth": 60,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                    >
                      
                    </Text>
                  </View>
                  <View
                    size={3}
                    style={
                      Object {
                        "borderWidth": 1,
                        "flex": 3,
                        "flexDirection": "column",
                      }
                    }
                  >
                    <View
                      accessibilityComponentType={undefined}
                      accessibilityLabel={undefined}
                      accessibilityTraits={undefined}
                      accessible={true}
                      collapsable={undefined}
                      hasTVPreferredFocus={undefined}
                      hitSlop={undefined}
                      isTVSelectable={true}
                      nativeID={undefined}
                      onLayout={undefined}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "backgroundColor": "#bebebe",
                          "height": "100%",
                          "opacity": 1,
                          "width": "100%",
                        }
                      }
                      testID={undefined}
                      tvParallaxProperties={undefined}
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                      />
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F5FCFF",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "red",
          "textAlign": "center",
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "green",
          "textAlign": "center",
        }
      }
    />
  </View>
</View>
`;
